src/main/scala/firrtl/passes/memlib/MemIR.scala

```diff
@@ -0,0 +1,43 @@
+// See LICENSE for license details.
+
+package firrtl.passes
+package memlib
+
+import firrtl._
+import firrtl.ir._
+import Utils.indent
+
+case class DefAnnotatedMemory(
+    info: Info,
+    name: String,
+    dataType: Type,
+    depth: Int,
+    writeLatency: Int,
+    readLatency: Int,
+    readers: Seq[String],
+    writers: Seq[String],
+    readwriters: Seq[String],
+    readUnderWrite: Option[String],
+    maskGran: Option[BigInt],
+    memRef: Option[String]
+    //pins: Seq[Pin],
+    ) extends Statement with IsDeclaration {
```

Comment:
You can't extend case classes and DefMemory is a case class. We could consider making built-in Firrtl IR nodes "DIY" case classes (which would then be extendable), but that is quite a bit of code to write since we'd have to reimplement all of the methods that case classes generate